lib/modules/versioning/rust-release-channel/index.ts

```diff
@@ -0,0 +1,403 @@
+    if (channel === 'nightly') {
+      return date !== undefined;
+    }
+
+    // Support versioned channels like `1.82.0` or `1.83.0-beta.5`
+    if (typeof channel === 'object') {
+      // Without patch, it's a range
+      if (channel.patch === undefined) {
+        return false;
+      }
+
+      // With beta without number, it's a range
+      return !channel.prerelease || channel.prerelease.number !== undefined;
+    }
+
+    return false;
+  }
+
+  isSingleVersion(input: string): boolean {
+    return this.isVersion(input);
+  }
+
+  isStable(version: string): boolean {
+    const parsed = parse(version);
+    if (!parsed) {
+      return false;
+    }
+
+    const { channel } = parsed;
+
+    // Channel names and ranges are not stable
+    if (typeof channel !== 'object') {
+      return false;
+    }
+
+    // Version without patch is a range
+    if (channel.patch === undefined) {
+      return false;
+    }
+
+    // Version with prerelease is not stable
+    return !channel.prerelease;
+  }
+
+  isCompatible(version: string, current?: string): boolean {
+    if (!current) {
+      return true;
+    }
+
+    const parsedVersion = parse(version);
+    const parsedCurrent = parse(current);
+    if (!parsedVersion || !parsedCurrent) {
+      return true;
+    }
+
+    const channelVersion = parsedVersion.channel;
+    const channelCurrent = parsedCurrent.channel;
+
+    // Nightly versions are compatible only with other nightlies
+    // Stable and beta versions are compatible with each other
+
+    const isVersionNightly = channelVersion === 'nightly';
+    const isCurrentNightly = channelCurrent === 'nightly';
+
+    return isVersionNightly === isCurrentNightly;
+  }
+
+  getMajor(version: string): number | null {
+    const parsed = parse(version);
+    if (!parsed) {
+      return null;
+    }
+
+    const { channel, date } = parsed;
+
+    // Nightly with date
+    if (channel === 'nightly' && date) {
+      // Rust 1.0.0 was released on 2015-05-15
+      const versionDate = new Date(date.year, date.month - 1, date.day);
+      return versionDate >= rust1Date ? 1 : 0;
+    }
+
+    // Version object
+    if (typeof channel === 'object') {
+      return channel.major;
+    }
+
+    // Default to major version 1 otherwise
+    return 1;
+  }
+
+  getMinor(version: string): number | null {
+    const parsed = parse(version);
+    if (!parsed) {
+      return null;
+    }
+
+    const { channel } = parsed;
+
+    // Only version objects have minor
+    if (typeof channel === 'object') {
+      return channel.minor;
+    }
+
+    return null;
+  }
+
+  getPatch(version: string): number | null {
+    const parsed = parse(version);
+    if (!parsed) {
+      return null;
+    }
+
+    const { channel } = parsed;
+
+    // Only version objects have patch
+    if (typeof channel === 'object') {
+      return channel.patch ?? null;
+    }
+
+    return null;
+  }
+
+  sortVersions(version: string, other: string): number {
```

Comment:
this should be called `compareVersions`

Reply:
are you sure? https://github.com/renovatebot/renovate/blob/42.42.2/lib/modules/versioning/types.ts#L114-L118 seems to disagree 🤔